compiler/rustc_middle/src/ty/util.rs

```diff
@@ -770,6 +770,39 @@ impl<'tcx> ty::TyS<'tcx> {
         }
     }
 
+    /// Checks if `ty` has has a significant drop
+    ///
+    /// Note that this implies that `ty` *might* have a destructor attached
+    /// if it returns false then then the type has a destructor on some type
```

Comment:
"even" if it returns false. If that's the case then the adt has been marked with insignificant_dtor...

~~then~~ then

Note that this method .... 2229 drop reorder migration analysis